### PR TITLE
new feature: import excel files, use sheets as sqlite tables; see #59916

### PR DESCRIPTION
**Pull request by phempel**

### CODE REVIEW

File: data2sqlite.py [hunk 1]
```diff
@@ -45,8 +45,9 @@
 
 import_parser=subparsers.add_parser('file', help="Add to Source from other files")
 import_parser.add_argument('--sqlite', nargs='*', default=[],                   help='Filename for SQLite Database')
-import_parser.add_argument('--XML', nargs='*', default=[],                      help='Filename for XML')
-import_parser.add_argument('--CSV', nargs='*', default=[],                      help='Filename for CSV')
+import_parser.add_argument('--XML', nargs='*', default=[],                      help='Filename(s) for XML')
+import_parser.add_argument('--CSV', nargs='*', default=[],                      help='Filename(s) for CSV')
+import_parser.add_argument('--excel', nargs='*', default=[],                    help='Filename(s) for Excel')
```
Comment (martinrode) [inline data2sqlite.py:50]: the help can list the supported formats

File: data2sqlite.py [hunk 2]
```diff
@@ -293,6 +294,14 @@
                 filename=csv_file,
                 )
 
+    if args.excel !=[]:
+        for csv_file in args.excel:
+            logging.info("Adding Excel file to Source")
+            extract.excel_to_source(
+                name=args.name,
+                filename=csv_file,
```
Comment (martinrode) [inline data2sqlite.py:302]: `csv_file` ?

**Review by martinrode — CHANGES_REQUESTED** [review]
only two minor things. looks good

File: data2sqlite.py (hunk 1 above)
Comment (phempel) [reply data2sqlite.py]: - I added the supported formats as they are supported by the library
- `ods` is not supported

File: data2sqlite.py (hunk 2 above)
Comment (phempel) [reply data2sqlite.py]: typo, fixed it

**Pull request merged**